core/chains/evm/client/node.go

```diff
@@ -86,6 +86,7 @@ type Node interface {
 	Close()
 
 	State() NodeState
+	LatestReceivedBlockNumber() int64
```

Comment:
I think it is important to fetch state and block number atomically - not just for efficiency, but for correctness too.

Reply:
On my TODO.

Reply:
LGTM other than this ^

Reply:
Done.